### PR TITLE
Add VVVF parser to read box arrays from string

### PR DESCRIPTION
**Pull request by ssr-yuki**
For the format proposed in https://github.com/smilerobotics/laser_filters/issues/1#issue-840502066

### CODE REVIEW

File: src/box_utils.cpp [hunk 1]
```diff
@@ -182,4 +182,90 @@ Box padBox(const Box& box, double padding)
...
+std::vector<std::vector<std::vector<float>>> parseVVVF(const std::string& input, std::string& error_return)
+{
+  std::stringstream input_ss(input);
+  int depth = 0;
+
+  std::vector<float> current_vector;
+  std::vector<std::vector<float>> current_vvf;
+  std::vector<std::vector<std::vector<float>>> result;
+
+  while (!!input_ss && !input_ss.eof())
```
Comment (OTL) [inline src/box_utils.cpp:195]: I didn't know how to use `!!`.
The purpose is to convert the `input_ss` into bool and reverse it?
Is this normal?

File: src/box_utils.cpp [hunk 2]
```diff
@@ -182,4 +182,90 @@ Box padBox(const Box& box, double padding)
...
+  {
+    switch (input_ss.peek())
+    {
+      case EOF:
+        break;
+      case '[':
+        depth++;
```
Comment (OTL) [inline src/box_utils.cpp]: Let's use ++depth.

File: src/box_utils.cpp [hunk 3]
```diff
@@ -182,4 +182,90 @@ Box padBox(const Box& box, double padding)
...
+        }
+        else if (depth > 3)
+        {
+          error_return = "Array depth greater than 3";
+          return result;
+        }
+        input_ss.get();
+        break;
+      case ']':
+        depth--;
```
Comment (OTL) [inline src/box_utils.cpp]: ditto

File: src/box_utils.cpp [hunk 4]
```diff
@@ -182,4 +182,90 @@ Box padBox(const Box& box, double padding)
...
+        if (depth != 3)
+        {
+          std::stringstream err_ss;
+          err_ss << "Numbers at depth " << depth << " . Char was '" << char(input_ss.peek()) << "'.";
+          error_return = err_ss.str();
+          return result;
+        }
+
+        // reads a float value
+        float value;
```
Comment (OTL) [inline src/box_utils.cpp]:
```c++
float value = 0.0f;
```

let's initialize all values.

File: src/box_utils.cpp (hunk 1 above)
Comment (ssr-yuki) [reply src/box_utils.cpp:195]: Thank you. This code is a copy of polygon_utils written by other developers. I will check it.
https://github.com/smilerobotics/laser_filters/blob/5777e4ee66ca1078e037591cdb24ad7e9cc4192b/src/polygon_utils.cpp#L88

Comment (ssr-yuki) [reply src/box_utils.cpp:195]: I think it is not a wrong representation.

 `!` operator returns `ture` if some errors occur in the stringstream, and the additional `!` reverses it as you guess. As a result, `!!` returns `true` if there exist no error.
https://en.cppreference.com/w/cpp/io/basic_ios/operator!

**Review by OTL — APPROVED** [review]

**Pull request merged**